Source/WebKit/NetworkProcess/NetworkProcess.h

```diff
@@ -506,6 +509,8 @@ class NetworkProcess : public AuxiliaryProcess, private DownloadManager::Client,
     HashMap<PAL::SessionID, std::unique_ptr<NetworkSession>> m_networkSessions;
     HashMap<PAL::SessionID, std::unique_ptr<WebCore::NetworkStorageSession>> m_networkStorageSessions;
 
+    HashMap<WebCore::ProcessIdentifier, HashSet<WebCore::RegistrableDomain>> m_allowedFirstPartiesForCookies;
```

Comment:
Why isn't this a HashSet on the NetworkConnectionToWebProcess object? Seems like a logical spot for it rather than having a global HashMap on the NetworkProcess class.
 
Is there a technical reason for this? The only one I could think of is if we get the first party for a connection *before* we construct the connection in some cases. No idea if this can happen in practice though.

Reply:
This does happen in practice, and I implemented it that way initially and it added a lot of complexity to do it that way and have code paths for if NetworkConnectionToWebProcess has and has not been created yet.  This was much more elegant.